### ISSUE TITLE
question/help with cmi formatting and auto kick commands

### ISSUE THREAD

**Issue by MrMcyeet** [Zrips/CMI#4447]
Alrighty, im PROBABLY doing this WAY more complicated than I need to be, but I had the bright idea to have different kick timers for different ingame ranks.

I *Thought* I was smart and started messing around with ConditionalCommands (like, I went as far as modifying the plugin to add PAPI support xD) and I came up with this wall of text for the "AutoKickCmds" in the CMI config. you probably know about the conditional commands plugin, if not its spigot page is here --> https://www.spigotmc.org/resources/conditionalcommands.14295/

Anyway, here is what I made for the AutoKickCmds 
```

  AutoKickCmds:
  - ccmd [playerName] if (-perm:group.d1-=0&-perm:group.d3-=0&-perm:group.d5-=0&-perm:group.d6-=0) do kick [playerName] Auto-kicked for AFK. Feel free to relog. -s
  - reflex delay 600 ccmd [playerName] if (-perm:group.d3-=1&-perm:group.d5-=0&-perm:group.d6-=0)&(-time_online->600000)&(%formatter_text_substring_2:3_{cmi_user_afk_for}%>8) do kick [playerName] Auto-kicked for AFK. Feel free to relog. -s
  - reflex delay 900 ccmd [playerName] if (-perm:group.d5-=1&-perm:group.d6-=0)&(-time_online->900000)&(%formatter_text_substring_2:4_{cmi_user_afk_for}%>13) do kick [playerName] Auto-kicked for AFK. Feel free to relog. -s
  - reflex delay 900 ccmd [playerName] if (-perm:group.d6-=1)&(-time_online->900000)&(%formatter_text_substring_2:4_{cmi_user_afk_for}%>13) do kick [playerName] Auto-kicked for AFK. Feel free to relog. -s

```

Im gonna break this down, so we can follow my though process with it. 
The first part(on most of them) is either a 10/15 minute delay, this is how im letting donors "bypass" the auto afk-kick for longer

The second part is my  `ccmd [playerName] if` this is needed for the conditional commands plugin

Next up is the conditions (im gonna only go over one as an example, but you should get it) 
`(-perm:group.d3-=1&-perm:group.d5-=0&-perm:group.d6-=0)&(-time_online->600000)&(%formatter_text_substring_2:3_{cmi_user_afk_for}%>8)` 
this does a few things. first, it checks that the player has `group.d3` and NOT `group.d5` or `group.d6` the reason im doing this is to create a distinction between the different rans. so d3 has a different timer than the ranks above it, so by checking what the highest permission that X rank should have, and verifying that they dont have any higher ranks, we can correctly target X rank
The next part of this section is just checking that they have been online for how long the delay at the beginning was, so we can make sure that they havent logged out since the command was called.
and the last part is this -->`%formatter_text_substring_2:3_{cmi_user_afk_for}%>13` and this took me FOREVER to come up with. basically it just looks at the `cmi_user_afk_for` variable, and verifies that the first number is greater than 8 (because the variable returns §615 §emin §620 §esec if a user has been afk for 15m 20s, the formatter placeholder just takes the cmi user afk for placeholder and grabs the 3rd and 4th character in it.) i do this to verify that they have been actually afk since the delay at the beginning. Since its a flat delay, they may have come back, since it was called, and without this, they would be kicked while playing.

The next part is 
`do kick [playerName] Auto-kicked for AFK. Feel free to relog. -s`
Pretty simple, `do` is just what we are telling ConditionalCommands to do if the conditions return true. the last part is just our kick command.

Now to my question. 
Does CMI do some formatting magic with its config? 

cause conditional commands keeps saying "Invalid syntax" when run through the cmi autoafk system, but when i run it through the console manually, it works fine.

so I did some digging, and found that the text editor I was using for CMI's config (Sublime text) has a setting called `wrap_width` which basically just decides when/where it should wrap a line of the file to the next line.
This is good, so I set this to 500, saved it, and closed the settings
Then I updated the CMI config to have each of the auto-kick-cmds in ONE line
![CMIconfigONmyDESKTOP](https://user-images.githubusercontent.com/68966344/97067318-c8471c00-1581-11eb-8228-538ca63480d5.JPG)

Then I saved it to the server, and did /cmi reload
Then I also saved a copy of the config to my desktop (This is the one i put on my desktop)
![CMIconfigONmyDESKTOP](https://user-images.githubusercontent.com/68966344/97067315-ba919680-1581-11eb-8d9c-d47de19f23f3.JPG)

After reopening the CMI config that is on the server AND desktop, and the one that was on the server was wrapped around just like it was before I made the changes, but the one I had saved onto my desktop HADDENT 
this is what was loaded, when I loaded it from the server
![CMIautoWRAPPEDconfig](https://user-images.githubusercontent.com/68966344/97067335-f3317000-1581-11eb-939e-a13ec384d524.JPG)

the reason I suspected that the auto-line wrapping was causing the issue is because that was the only difference  I saw when executing the command from cmi auto-kick commands and console.
And I also suspected this, because if i were to wrap each of the commands in quotes, the quotes get taken out automatically?

So basically what im asking, is 
A) Is there a simpler way to do my kick-by-rank system
B) Does CMI automatically wrap the text of the config around to the next line if its longer than a certain length?
C) If B is true, can we have that configurable via the config? i.e. adding
```
#Dont touch unless you are really smart
AutoWrapping: true
AutoWrappingLenght: 500
```
to the config

**Comment by CloudeLecaw** [comment]
what  you wrote down there will not work because thats not how the autokickcommands works
The autokic command is scanned once just when the player is afk for time X (set in config) and execute the command after ....what usually is a kick command

**Comment by Zrips** [comment]
@MrMcyeet 
Some what overcomplication, will add option to define kick out timer with basic cmi.command.afk.kickOutIn.[seconds] permission node, so you can avoid using all that.
